File: php/slim-doctrine/app/src/com/sprint/sms/api/access/MenuAccess.php
<?php
namespace App\com\sprint\sms\api\access;

use \App\com\sprint\sms\api\support\FormatSupport;

class MenuAccess implements FormatSupport
{	
	private $id;
	
	private $title;
	
	private $link;
	
	private $icon;
	
	private $description;
	
	private $parent;
	
	private $children = array();
	
	private $action = array();
	
	public function setId($id) {
		$this->id = $id;
	}
	
	public function getId() {
		return $this->id;
	}
	
	public function setTitle($title) {
		$this->title = $title;
	}
	
	public function getTitle() {
		return $this->title;
	}	
	
	public function setLink($link) {
		$this->link = $link;
	}
	
	public function getLink() {
		return $this->link;
	}
	
	public function setIcon($icon) {
		$this->icon = $icon;
	}
	
	public function getIcon() {
		return $this->icon;
	}
	
	public function setDescription($description) {
		$this->description = $description;
	}
	
	public function getDescription() {
		return $this->description;
	}
	
	public function setParent(MenuAccess $parent = null) {
		$this->parent = $parent;
	}
	
	public function getParent() {
		return $this->parent;
	}	
	
	public function setChildren(array $children = array()) {
		$this->children = $children;
	}
	
	public function getChildren() {
		return $this->children;
	}	
	
	public function setAction(array $action = array()) {
		$this->action = $action;
	}
	
	public function getAction() {
		return $this->action;
	}	
	
	public function toFormatObject() {
		$o = array();
		$o["id"] = $this->id;
		$o["title"] = $this->title;
		$o["link"] = $this->link;	
		$o["icon"] = $this->icon;	
		$o["description"] = $this->description;	
		$o["parent"] = isset($this->parent) ? array("id" => $this->parent->getId()) : null;	
		$o["action"] = $this->action;
		if (isset($this->children)) {
			$o["children"] = array();
			for ($i = 0; $i < count($this->children); $i++) {				
				$o["children"][$i] = $this->children[$i]->toFormatObject();
			}
		}
		return $o;
	}	
}

File: java/spring-hibernate/src/main/java/com/sprint/sms/api/access/MenuAccess.java
package com.sprint.sms.api.access;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

import com.ideahut.common2.annotation.IdhFormatter;

@IdhFormatter
public class MenuAccess implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = 8822272953562229833L;
	
	private Long id;
	
	private String title;
	
	private String link;
	
	private String icon;
	
	private String description;
	
	private MenuAccess parent;
	
	private List<MenuAccess> children = new ArrayList<MenuAccess>();
	
	private Set<String> action = new HashSet<String>();	
	

	@IdhFormatter
	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	@IdhFormatter
	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}

	@IdhFormatter
	public String getLink() {
		return link;
	}

	public void setLink(String link) {
		this.link = link;
	}

	@IdhFormatter
	public String getIcon() {
		return icon;
	}

	public void setIcon(String icon) {
		this.icon = icon;
	}

	@IdhFormatter
	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}
	
	@IdhFormatter(field = "id")
	public MenuAccess getParent() {
		return parent;
	}

	public void setParent(MenuAccess parent) {
		this.parent = parent;
	}

	@IdhFormatter
	public List<MenuAccess> getChildren() {
		return children;
	}

	public void setChildren(List<MenuAccess> children) {
		this.children = children;
	}

	@IdhFormatter
	public Set<String> getAction() {
		return action;
	}

	public void setAction(Set<String> action) {
		this.action = action;
	}
	
}
